src/screens/settings/components/SettingsContainer.js

```diff
@@ -0,0 +1,230 @@
+import React, { useState, useEffect, useCallback, useContext } from 'react';
+
+// Components
+import { StatusBar, Alert, StyleSheet, View } from 'react-native';
+import {
+  Container,
+  Header,
+  Title,
+  Content,
+  Footer,
+  FooterTab,
+  Button,
+  Left,
+  Right,
+  Body,
+  Icon,
+  Text,
+  Form,
+  Picker,
+  H1,
+  H3,
+} from 'native-base';
+import NameBar from './NameBar';
+
+// Utils
+import {
+  getUser,
+  setUser,
+  clearWordList,
+  getUserWordList,
+  getAllUsers,
+} from '../../../utils/storage';
+
+// Context
+import UserContext from '../../../contexts/UserContext';
+
+// Styles
+const styles = StyleSheet.create({
+  wordListContainer: {
+    display: 'flex',
+    flexDirection: 'row',
+    padding: 5,
+  },
+  wordList: {
+    flex: 5,
+    flexDirection: 'row',
+  },
+  refresh: {
+    flex: 1,
+    flexDirection: 'row',
+  },
+  wordLevel: {
+    padding: 15,
+  },
+});
+
+// Vars
+const NEW_USER = 'New User';
+
+const updateData = (inputText, setName) => {
+  setName(inputText);
+  setUser(inputText);
+};
+
+const getData = async () => getUser();
+
+const clearWords = async () => {
+  clearWordList();
+};
+
+const handleClearButtonPress = () => {
+  Alert.alert(
+    'Are you sure you want to delete all of your words?',
+    'This action is irreversible. You will lose all of your word data.',
+    [
+      {
+        text: 'Cancel',
+        onPress: () => {},
+        style: 'cancel',
+      },
+      { text: 'Delete All Words', onPress: clearWords, style: 'danger' },
+    ],
+  );
+};
+
+const getUsers = async () => {
+  const users = await getAllUsers();
+  return users;
+};
+
+const SettingsContainer = () => {
+  const [name, setName] = useContext(UserContext);
+  const [inputText, setInputText] = useState('');
+  const [userData, setUserData] = useState({});
+  const [users, setUsers] = useState([]);
+  const [inputVisible, setInputVisible] = useState(false);
+
+  const refreshList = useCallback(() => {
+    getUserWordList(name).then(setUserData);
+  }, [name]);
```

Comment:
I _think_ this use of useCallback is unnecessary, but I haven't used it very much. Was reading https://kentcdodds.com/blog/usememo-and-usecallback and it seems like this would work just as well as a typical function.

Reply:
Yeah I'm not comfortable with useCallback either, so I'd be open to getting rid of it. I took this as a suggestion from my linter

![image](https://user-images.githubusercontent.com/27874045/88680330-5fedb900-d0be-11ea-93a5-c5c396fe9940.png)

Reply:
Innnteresting. Linting hooks always seems to give me odd advice — like, it'll tell me never to use empty arrays when in some cases they're perfectly fine. I suppose there's not much a difference here 🤷

Reply:
This makes sense. If the value is used as a dependency to another effect then making sure it updates as little as necessary is important.